test/rateLimiters/slidingWindowCounter.test.ts

```diff
@@ -0,0 +1,476 @@
+            // previousTokens (in previous fixed window): 8
+            const count = await getWindowFromClient(client, user4);
+            expect(count.currentTokens).toBe(0);
+            expect(count.previousTokens).toBe(initRequest);
+        });
+
+        test('rolling window at 25% blocks requests over allowed limit set by formula', async () => {
+            // 25% of rolling window present in previous fixed window
+            // 1.75*60000 = 105000 (time after initial fixedWindowStart
+            // to set rolling window at 25% of previous fixed window)
+
+            // to set initial fixedWindowStart
+            await setTokenCountInClient(client, user4, 0, null, timestamp);
+
+            const initRequest = 8;
+
+            // large request at very end of first fixed window
+            await limiter.processRequest(user4, timestamp + WINDOW_SIZE, initRequest);
+
+            // 9 + 8 * .25 = 11, over capacity (request should be blocked)
+            expect(
+                (await limiter.processRequest(user4, timestamp + WINDOW_SIZE * 1.75, 9)).tokens
+            ).toBe(10);
+
+            // currentTokens (in current fixed window): 0
+            // previousTokens (in previous fixed window): 8
+            const count = await getWindowFromClient(client, user4);
+            expect(count.currentTokens).toBe(4);
+            expect(count.previousTokens).toBe(initRequest);
+        });
+    });
+
+    describe('SlidingWindowCounter functions as expected', () => {
+        afterEach(() => {
+            client.flushall();
+        });
+
+        test('allows user to consume current allotment of tokens', async () => {
+            // "free requests"
+            expect((await limiter.processRequest(user1, timestamp, 0)).success).toBe(true);
+            // Test 1 token requested
+            expect((await limiter.processRequest(user1, timestamp, 1)).success).toBe(true);
+            // Test < CAPACITY tokens requested
+            expect((await limiter.processRequest(user2, timestamp, CAPACITY - 1)).success).toBe(
+                true
+            );
+            // <= CAPACITY tokens requested
+            expect((await limiter.processRequest(user3, timestamp, CAPACITY)).success).toBe(true);
+        });
+
+        test('blocks exceeding requests over token allotment', async () => {
+            // Test > capacity tokens requested
+            expect((await limiter.processRequest(user1, timestamp, CAPACITY + 1)).success).toBe(
+                false
+            );
+
+            // Fill up user 1's window
+            const value: RedisWindow = {
+                currentTokens: 10,
+                previousTokens: null,
+                fixedWindowStart: timestamp,
+            };
+            await client.set(user1, JSON.stringify(value));
+
+            // window is full. Shouldn't be allowed to take 1 token
+            expect((await limiter.processRequest(user1, timestamp, 1)).success).toBe(false);
```

Comment:
Shouldn't `timestamp` should be `timestamp + 60000` otherwise the request is process at the beginning of the previous window?